### PR TITLE
rpm: install Linux kernel headers for btrfs on CentOS 7

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/docker/packaging/pull/69#issuecomment-1374737317
- relates to https://github.com/moby/moby/pull/44761
- relates to https://github.com/moby/moby/pull/44762

### CODE REVIEW

**Comment by thaJeztah** [comment]
Tried to use `sed -i 's/$basearch/x86_64/' /etc/yum.repos.d/elrepo.repo` as a quick hack to make it use x86, but that didn't seem to work either 🤔

**Comment by thaJeztah** [comment]
Looking at one of the mirrors, the package is named `kernel-lt-headers-5.4.227-1.el7.elrepo.x86_64.rpm` (so not a `no arch` 🤔); http://mirrors.colocall.net/elrepo/kernel/el7/x86_64/RPMS/

**Comment by crazy-max** [comment]
> Looking at one of the mirrors, the package is named `kernel-lt-headers-5.4.227-1.el7.elrepo.x86_64.rpm` (so not a `no arch` 🤔); http://mirrors.colocall.net/elrepo/kernel/el7/x86_64/RPMS/

Oh right didn't catch that!

File: rpm/centos-7/Dockerfile [hunk 1]
```diff
@@ -21,6 +21,17 @@ ENV SUITE=${SUITE}
...
+
+# Install Linux kernel headers for btrfs support.
+# See https://github.com/moby/moby/pull/44761 and https://github.com/moby/moby/pull/44762,
+# See https://www.kernel.org/doc/Documentation/kbuild/headers_install.txt
+RUN yum install -y make gcc \
+ && curl -fsSL -O https://github.com/torvalds/linux/archive/refs/tags/v4.12.tar.gz \
+ && tar -xvzf ./v4.12.tar.gz \
+ && rm v4.12.tar.gz \
+ && cd linux-4.12/ \
+ && make headers_install ARCH=arm64 INSTALL_HDR_PATH=/usr
```
Comment (neersighted) [inline rpm/centos-7/Dockerfile:33]: It seems like we shouldn't be setting ARCH here, should we? I don't quite see how this Dockerfile is ARM-specific.

**Comment by thaJeztah** [comment]
We'll probably go for https://github.com/docker/docker-ce-packaging/pull/811 instead

**Pull request closed without merging**